### PR TITLE
00003 

### PR DESCRIPTION
**Pull request by Sephirot92**
Added Service, Service Implementation class and basic Lawsuit class in version 0.0.1.

### CODE REVIEW

File: src/main/java/com/leggeart/legge/Lawsuit.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+import java.util.Date;
+
+/**
+ Class responsible for handling Lawsuit object. Contains all data connected
+ to a case.
+
+ @version 0.0.1
+ @author LC
+ */
+@Entity
```
Comment (ganzes) [inline src/main/java/com/leggeart/legge/Lawsuit.java:15]: @Entity(name = "LAWSUITS")
you don't have to add adnotations @Table if you have already @Entity

File: src/main/java/com/leggeart/legge/Lawsuit.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+@Table(name="LAWSUITS")
+public class Lawsuit {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private int id;
```
Comment (ganzes) [inline src/main/java/com/leggeart/legge/Lawsuit.java:21]: @Column(name="ID")

File: src/main/java/com/leggeart/legge/Lawsuit.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    @Column(name = "PLAINTIFFREPRESENTATIVE")
+    private String plaintiffRepresentative;
+
+    @Column(name = "DEFENDANT", nullable = false)
+    private String defendant;
+
+    @Column(name = "defendantRepresentative")
+    private String defendantRepresentative;
+
+    @Column (name = "THEME", nullable = false)
```
Comment (ganzes) [inline src/main/java/com/leggeart/legge/Lawsuit.java:41]: IMO motive should be better than theme

File: src/main/java/com/leggeart/legge/Lawsuit.java [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        this.signature = signature;
+        this.court = court;
+        this.plaintiff = plaintiff;
+        this.plaintiffRepresentative = plaintiffRepresentative;
+        this.defendant = defendant;
+        this.defendantRepresentative = defendantRepresentative;
+        this.theme = theme;
+        this.salary = salary;
+        this.wasSalaryPayed = wasSalaryPayed;
+        this.begginingDate = new Date();
```
Comment (ganzes) [inline src/main/java/com/leggeart/legge/Lawsuit.java:71]: this.begginingDate = new Date();
>        this.begginingDate = begginingDate;
you don't have to initiate object new Date() here imho, or you had to do exactly in this manner this.endingDate = endingDate;

File: src/main/java/com/leggeart/legge/LawsuitRepository.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
+package com.leggeart.legge;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+import javax.transaction.Transactional;
+import java.util.List;
+
+/**
+ Dao class to handle Lawsuit class.
```
Comment (ganzes) [inline src/main/java/com/leggeart/legge/LawsuitRepository.java:10]: DAO is not a repository 
https://www.baeldung.com/java-dao-vs-repository#:~:text=DAO%20works%20as%20a%20data,be%20implemented%20using%20a%20repository.

File: src/main/java/com/leggeart/legge/LawsuitRepository.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+ @version 0.0.1
+ @author LC
+ */
+@Transactional
+@Repository
+public interface LawsuitRepository extends JpaRepository<Lawsuit, Integer> {
```
Comment (ganzes) [inline src/main/java/com/leggeart/legge/LawsuitRepository.java:17]: public interface LawsuitRepository extends JpaRepository<Lawsuit, Integer> {
> public interface LawsuitRepository extends JpaRepository<Lawsuit, Long>

File: src/main/java/com/leggeart/legge/Lawsuit.java (hunk 2 above)
Comment (ganzes) [inline src/main/java/com/leggeart/legge/Lawsuit.java:21]: i would make id as a long;

File: src/main/java/com/leggeart/legge/LawsuitService.java [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+import java.util.List;
+import java.util.Optional;
+
+
+/**
+ Lawsuit service interface.
+ @version 0.0.1
+ @author LC
+ */
+public interface LawsuitService {
```
Comment (ganzes) [inline src/main/java/com/leggeart/legge/LawsuitService.java:12]: @Service
we are using Spring, annotations ad the top of every class is strongly recommended.

File: src/main/java/com/leggeart/legge/LawsuitServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,36 @@
+package com.leggeart.legge;
+
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.List;
+import java.util.Optional;
+
+public class LawsuitServiceImpl implements  LawsuitService{
```
Comment (ganzes) [inline src/main/java/com/leggeart/legge/LawsuitServiceImpl.java:8]: IMO you don't have to add another class, you could have all those methods in LawsuitService.
Also, you need to add annotations at the top - @Service

**Review by ganzes — CHANGES_REQUESTED** [review]

**Comment by Sephirot92** [comment]
Can you see those changes?

**Review by ganzes — APPROVED** [review]

**Pull request merged**